cajita/src/Cajita_SparseDimPartitioner.hpp

```diff
@@ -33,10 +33,14 @@ namespace Cajita
   \tparam Device Kokkos device type.
   \tparam CellPerTileDim Cells per tile per dimension.
 */
-template <typename Device, unsigned long long CellPerTileDim = 4>
-class SparseDimPartitioner : public BlockPartitioner<3>
+template <typename Device, unsigned long long CellPerTileDim = 4,
+          std::size_t NumSpaceDim = 3>
+class SparseDimPartitioner : public BlockPartitioner<NumSpaceDim>
```

Comment:
The preparation for dimension generalization is nice, but I think we need a note in the doxygen that this only supports 3D